climpyrical/tests/test_gridding.py

```diff
@@ -0,0 +1,301 @@
+from climpyrical.gridding import (
+    check_input_coords,
+    check_coords_are_flattened,
+    check_transform_coords_inputs,
+    flatten_coords,
+    transform_coords,
+    check_find_nearest_index_inputs,
+    check_find_element_wise_nearest_pos_inputs,
+    check_find_nearest_value_inputs,
+    find_nearest_index,
+    find_element_wise_nearest_pos,
+    find_nearest_index_value,
+)
+import pytest
+import numpy as np
+
+# create pseudo grids with expected dimension and ranges
+xi = np.linspace(-33.8800048828125, 33.8800048828125, 155)
+yi = np.linspace(-28.59999656677246, 28.15999984741211, 130)
+xext, yext = np.tile(xi, yi.size), np.repeat(yi, xi.shape)
+xext_bad, yext_bad = np.repeat(xi, yi.size), np.tile(yi, xi.shape)
+
+
+@pytest.mark.parametrize(
+    "x,y,passed",
+    [
+        ("x", np.linspace(-24, 24, 155), False),
+        (np.linspace(-24, 24, 155), "y", False),
+        (
+            np.linspace(0, 10, 30),
+            np.linspace(-28.59999656677246, 28.15999984741211, 130),
+            False,
+        ),
+        (
+            np.linspace(-33.8800048828125, 33.8800048828125, 155),
+            np.linspace(0, 10, 130),
+            False,
+        ),
+        (xi, yi, True),
+        (xext, yext, True),
+    ],
+)
+def test_check_input_coords(x, y, passed):
+    if passed:
+        assert check_input_coords(x, y)
+    else:
+        with pytest.raises((ValueError, TypeError)):
```

Comment:
Suggest you parametrize which error you expect to be raised in each case.

Reply:
I knew there was a better way to do that. Thanks. That's an answer to a question I've been meaning to ask but hasn't come up yet.